packaging_automation/package_cloud_statistics_collector.py

```diff
@@ -81,48 +106,105 @@ def fetch_and_save_package_cloud_stats(db_params: DbParams, package_cloud_api_to
             page_index = page_index + parallel_count
         else:
             break
-        fetch_and_save_package_stats_for_package_list(package_info_list, package_cloud_api_token, session,
-                                                      save_records_with_download_count_zero, repo_name)
+        for package_info in package_info_list:
+            fetch_and_save_package_download_details(package_info, package_cloud_api_token, session,
+                                                    repo_name)
+            fetch_and_save_package_stats(package_info, package_cloud_api_token, session,
+                                         save_records_with_download_count_zero, repo_name)
+
+            session.commit()
 
-        session.commit()
     end = time.time()
 
     print("Elapsed Time in seconds: " + str(end - start))
 
 
-def fetch_and_save_package_stats_for_package_list(package_info_list: List[Any], package_cloud_api_token: str, session,
-                                                  save_records_with_download_count_zero: bool,
-                                                  repo_name: PackageCloudRepos):
+def fetch_and_save_package_stats(package_info, package_cloud_api_token: str, session,
+                                 save_records_with_download_count_zero: bool,
+                                 repo_name: PackageCloudRepos):
     '''Gets and saves the package statistics of the given packages'''
-    for package_info in package_info_list:
-
+    request_result = stat_get_request(
+        package_statistics_request_address(package_cloud_api_token, package_info['downloads_series_url']),
+        RequestType.package_cloud_download_series_query, session)
+    if request_result.status_code != HTTPStatus.OK:
+        print(f"Error while getting package stat for package {package_info['filename']}")
```

Comment:
We error here but the packages we previously added to session will still be committed. Is this correct?
I think we should not commit them either.

Reply:
Fixed threw exception